### PR TITLE
TrainsLogger -> ClearMLLogger

### LINKED ISSUE #1553: Rename Trains to ClearML
## 🚀 Feature

According to https://www.clear.ml/blog/announcing-clearml let's rename our `TrainsLogger` and all related examples and mentions of trains to clearml while **keeping aliases** like [`trains_logger` module](https://github.com/pytorch/ignite/blob/master/ignite/contrib/handlers/trains_logger.py) and classes `TrainsLogger`, `TrainsSaver` for backward compatibility:

```
TrainsLogger -> ClearMLLogger
TrainsSaver -> ClearMLSaver

ignite/contrib/handlers/trains_logger.py -> ignite/contrib/handlers/clearml_logger.py 
examples/contrib/mnist/mnist_with_trains_logger.py -> examples/contrib/mnist/mnist_with_clearml_logger.py
tests/ignite/contrib/handlers/test_trains_logger.py -> tests/ignite/contrib/handlers/test_clearml_logger.py
examples/references/segmentation/pascal_voc2012/assets/trains_dashboard.png -> let's keep it as is
examples/references/classification/imagenet/experiments/trains/requirements.txt -> examples/references/classification/imagenet/experiments/clearml/requirements.txt
examples/references/classification/imagenet/NOTES_Trains.md -> examples/references/classification/imagenet/NOTES_ClearML.md 
examples/references/segmentation/pascal_voc2012/experiments/trains/requirements.txt -> examples/references/segmentation/pascal_voc2012/experiments/clearml/requirements.txt 
examples/references/segmentation/pascal_voc2012/NOTES_Trains.md -> examples/references/segmentation/pascal_voc2012/NOTES_ClearML.md
```

cc @bmartinn @jkhenning for visibililty

### PR DESCRIPTION
**Pull request by ydcjeff**
Fixes #1553 

Description: Renamed `TrainsLogger` to `ClearMLLogger` in all files found. Let me know if I miss something.

Check list:
* [x] New tests are added (if a new feature is added)
* [ ] New doc strings: description and/or example code are in RST format
* [x] Documentation is updated (if required)
https://deploy-preview-1557--pytorch-ignite-preview.netlify.app/contrib/handlers.html#clearml-logger
https://deploy-preview-1557--pytorch-ignite-preview.netlify.app/contrib/engines.html#ignite.contrib.engines.common.setup_trains_logging

### CODE REVIEW

File: ignite/contrib/engines/common.py [hunk 1]
```diff
@@ -537,13 +538,26 @@ def setup_trains_logging(
...
-        :class:`~ignite.contrib.handlers.trains_logger.TrainsLogger`
+        :class:`~ignite.contrib.handlers.clearml_logger.ClearMLLogger`
     """
-    logger = TrainsLogger(**kwargs)
+    logger = ClearMLLogger(**kwargs)
     _setup_logging(logger, trainer, optimizers, evaluators, log_every_iters)
     return logger
 
 
+def setup_trains_logging(
```
Comment (ydcjeff) [inline ignite/contrib/engines/common.py:548]: for backward compatibility

File: ignite/contrib/handlers/clearml_logger.py [hunk 2]
```diff
@@ -0,0 +1,803 @@
...
+    BaseWeightsHistHandler,
+    BaseWeightsScalarHandler,
+)
+from ignite.engine import Engine, Events
+from ignite.handlers import global_step_from_engine
+from ignite.handlers.checkpoint import DiskSaver
+
+__all__ = [
+    "ClearMLLogger",
+    "ClearMLSaver",
```
Comment (ydcjeff) [inline ignite/contrib/handlers/clearml_logger.py:28]: The actual change is in this file
all `trains` -> `clearml`

File: ignite/contrib/handlers/trains_logger.py [hunk 3]
```diff
@@ -1,802 +1,9 @@
...
-        super(TrainsSaver, self).remove(filename)
-        for slots in self._checkpoint_slots.values():
-            try:
-                slots[slots.index(filename)] = None
-            except ValueError:
-                pass
-            else:
-                break
+TrainsLogger = ClearMLLogger
+TrainsSaver = ClearMLSaver
```
Comment (ydcjeff) [inline ignite/contrib/handlers/trains_logger.py:9]: keeping alias

File: mypy.ini [hunk 4]
```diff
@@ -24,6 +24,9 @@ warn_unreachable = False
 warn_unused_configs = True
 warn_unused_ignores = True
 
+[mypy-clearml.*]
```
Comment (ydcjeff) [inline mypy.ini:27]: mypy change

File: tests/ignite/contrib/engines/test_common.py [hunk 5]
```diff
@@ -471,29 +472,49 @@ def test_setup_wandb_logging(dirname):
...
+            setup_logging_fn=setup_trains_logging,
+            kwargs_dict={},
+            output_handler_cls=handlers.clearml_logger.OutputHandler,
+            opt_params_handler_cls=handlers.clearml_logger.OptimizerParamsHandler,
+            with_eval=True,
+            with_optim=True,
+        )
+        clearml_logger.close()
+
+    with pytest.warns(UserWarning, match="setup_trains_logging was renamed to setup_clearml_logging"):
```
Comment (ydcjeff) [inline tests/ignite/contrib/engines/test_common.py:508]: BC check for `setup_trains_logging`

File: tests/ignite/contrib/handlers/test_clearml_logger.py [hunk 6]
```diff
@@ -3,13 +3,14 @@
...
 import torch
-import trains
-from trains.binding.frameworks import WeightsFileHandler
-from trains.model import Framework
+from clearml.binding.frameworks import WeightsFileHandler
+from clearml.model import Framework
 
 import ignite.distributed as idist
+from ignite.contrib.handlers.clearml_logger import *
 from ignite.contrib.handlers.trains_logger import *
```
Comment (ydcjeff) [inline tests/ignite/contrib/handlers/test_clearml_logger.py:14]: how shall we test aliases, `TrainsLogger` and `TrainsSaver`?

**Comment by sdesrozis** [comment]
@ydcjeff thank you very much for this huge work. Although I need some time to review it... I do it as soon as possible !!

**Comment by ydcjeff** [comment]
> @ydcjeff thank you very much for this huge work. Although I need some time to review it... I do it as soon as possible !!

You can take time to review it. Since most of the changes are changing `trains` to `clearml`, I don't split into small PRs.

File: tests/ignite/contrib/handlers/test_clearml_logger.py (hunk 6 above)
Comment (vfdev-5) [reply tests/ignite/contrib/handlers/test_clearml_logger.py:14]: Just creation of an instance and checking that it is a new class:
```python
    with pytest.warns(UserWarning, match="ClearMLSaver: running in bypass mode"):
        TrainsLogger.set_bypass_mode(True)
        logger = TrainsLogger(output_uri=dirname)
        assert isinstance(logger, ClearMLLogger)
```

**Review by vfdev-5 — COMMENTED** [review]
Thanks a lot for the PR ! 
I checked it and it looks good. I added the answer to your question.

**Review by sdesrozis — COMMENTED** [review]
It looks good for me ! Just let me until tomorrow to approve. Very well done !

**Review by sdesrozis — APPROVED** [review]
Looks good !! Thank you !

**Review by vfdev-5 — APPROVED** [review]
Thanks @ydcjeff !

**Pull request merged**